### DISCUSSION TITLE
Kong DB connection

### DISCUSSION THREAD

**Discussion by RagunathInGIT** [Kong/kong#7603] [Help] [answered]
Team - We are running  Kong on K8s backed with postgres DB. Does Kong hits DB for every calls from consumers to validate OAuth token, Basic Auth  / Key Auth creds..

**Comment by kikito** [comment, marked as answer]
Hello, it depends on the plugin.

Some plugins require constant access to the database (ratelimiting on certain policies) while others will cache things heavily and only need access to the database "the first time they see something" or when the cache expires or gets invalidated.

Oauth tokens need an active database at all times. Tokens need to be generated in and removed from the database constantly.

Basic Auth and Key Auth credentials will typically only hit the database the first time each worker "sees" a new consumer, and then use the cache afterwards.
